pkg/adaptation/adaptation.go

```diff
@@ -363,6 +380,40 @@ func (r *Adaptation) updateContainers(ctx context.Context, req []*ContainerUpdat
 	return r.updateFn(ctx, req)
 }
 
+// Validate requested container adjustments.
+func (r *Adaptation) validateContainerAdjustment(ctx context.Context, req *ValidateContainerAdjustmentRequest, result *result) (*CreateContainerResponse, error) {
+	rpl := result.createContainerResponse()
+
+	if req == nil || len(r.validators) == 0 {
+		return rpl, nil
+	}
+
+	req.AddResponse(rpl)
+	req.AddOwners(result.owners)
+
+	errors := make(chan error, len(r.validators))
+	wg := sync.WaitGroup{}
+
+	for _, p := range r.validators {
+		wg.Add(1)
+		go func(p *plugin) {
```

Comment:
Just a note for the future that we might want to revisit this slightly and think about whether we want to limit the concurrency a bit; might become an issue if there are many validator plugins.  No need to change as part of this PR.

Reply:
> Are there any places where we might need to update comments to explain that validation happens in parallel? (Just wondering if customers might try creating plugins with low `plugin_idx` and wonder why they don't receive priority)

@chrishenzie Yes, I think it will be a good idea to emphasize in the documentation that during the validation phase there is no 'plugin ordering by index' as validating plugins are run in parallel.

But regarding users wondering "why they don't get priority", I think they should not. Any presumed ordering/priority would be relevant only if it was possible to force validation both to fail and to succeed. But our semantics is deliberately such that the latter is not possible.

Reply:
true and users of the api may want ordering regardless of our scheme...   E.g. I will only approve after ___ approval, if my check is also valid.  Allocate claim foo before claim foo'...

Reply:
typically one would have a set operating in parallel at a level where the higher levels must finish first?  built-in, set-a.. set b with a dependency in set-a..

Reply:
> E.g. I will only approve after ___ approval, if my check is also valid

The validations are a logical AND, so if any fails the container is rejected.  I don't think you can model "I will only approve after $foo approval" since if $foo rejects the container is rejected anyway, and if $foo succeeds then you'd also succeed your validation?